swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -280,12 +281,27 @@ private SubscriptionEntity createSubscriptionForContract(
     var subscription = new SubscriptionEntity();
     mapper.mapContractEntityToSubscriptionEntity(subscription, contract);
     measurementMetricIdTransformer.translateContractMetricIdsToSubscriptionMetricIds(subscription);
+    if (subscription.getSubscriptionMeasurements().size() != contract.getMetrics().size()) {
```

Comment:
Would it bypass the below condition?
If we have subscription with Sockets and Cores then Contract with Sockets and Instance-hours with this code it won't remove the metric id that are not matching. Correct?

Reply:
At that point in the code the unsupported metricId would be removed already from subscriptions and this is merely checking the contract metrics and making sure that it reflects same metrics as the subscription_measurements